### ISSUE TITLE
Session defaults => 'database' throws exception on PHP 7.2

### ISSUE THREAD

**Issue by piotrpoz** [cakephp/cakephp#12434]
When configuring session details in config/app.php, using defaults 'database', a runtime exception is being thrown because of incorrect defaults values.

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.6.10
* Platform and Target: Apache 2.4.29, PHP 7.2.8

### What you did
A following session, running fine on  PHP 7.0.22, breaks after upgrading to PHP 7.2.8:

```
'Session' => [
        'defaults' => 'database',
        'cookie' => 'cookie',
        'timeout' => 0,
        'cookieTimeout' => 0,
        'autoRegenerate' => true,
        'checkAgent' => false,
        'ini' => [
            'session.cookie_lifetime' => 180
        ]
    ],
```

### What happened
It emits following warning:
```
Warning (4096): ini_set(): Cannot set 'user' save handler by ini_set() or session_module_name()
```
as well as throwing:
```
Error: [RuntimeException] Unable to configure the session, setting session.save_handler failed. #0 /var/www/cake3-app/vendor/cakephp/cakephp/src/Network/Session.php(214)
```

### What you expected to happen
A session should be configured and running properly.

### How to fix
Removing ```'defaults'  => 'database'``` fixes the issue. The problem lies in file
```vendor/cakephp/cakephp/src/Network/Session.php:166 ``` These defaults should not set ```'user'``` as session hanlder as it is being checked in ```vendor/cakephp/cakephp/src/Network/Session.php:113 ```

**Comment by ADmad** [comment]
The line numbers you mention don't correspond to current `Session.php` on `master` branch.

Are you sure you are using Cake 3.6.10? Please provide the output of 
`composer show cakephp/cakephp`.

**Comment by ADmad** [comment]
Duplicate of #11628.

The issue has already been fixed. Please ensure you are actually using the latest release.

**Comment by dereuromark** [comment]
The linked PR was part of 3.5.11 already. Not sure if that here is sth new. But let's wait on the feedback.

**Comment by piotrpoz** [comment]
@ADmad You are right. Looks like I've spent 3hrs chasing the witches. Can't explain it, looks like composer update misuse. Thanks anyway!
